src/frontend/src/handler/create_table.rs

```diff
@@ -1714,6 +1714,56 @@ pub async fn create_iceberg_engine_table(
 
     sink_with.insert("is_exactly_once".to_owned(), "true".to_owned());
 
+    if let Some(enable_compaction) = handler_args.with_options.get("enable_compaction") {
```

Comment:
Here are two issues:  
1. This logic is implemented in a verbose way. We can encapsulate the logic for checking `enable_compaction` within `WithOptions` to reduce if branches.  
2. When `enable_compaction == None`, we don't need to check the license and should set `enable_compaction=false`.